services/openvpn/service/manager.go

```diff
@@ -91,23 +98,27 @@ func (m *Manager) Serve(providerID identity.Identity) (err error) {
 		return errors.Wrap(err, "failed to add NAT forwarding rule")
 	}
 
-	m.releasePorts = m.mapPort()
+	m.port, err = m.ports.Acquire(m.serviceOptions.Protocol)
+	if err != nil {
+		return errors.Wrap(err, "failed to acquire an unused port")
+	}
+	m.releasePorts = m.mapPort(m.port.Num())
```

Comment:
can still get rid of the m.port and m.releasePorts. We'll have less port variables, makes it easier to understand as I'm a bit confused now by the naming, especially this part: 
```golang
m.releasePorts = m.mapPort(m.port.Num())
```

So i suggest the following changes:

```golang
port, err := m.ports.Acquire(m.serviceOptions.Protocol)
if err != nil {
	return errors.Wrap(err, "failed to acquire an unused port")
}
defer m.mapPort(m.port.Num())()
```

then for the firewall part:

```golang
log.Info(logPrefix, "starting openvpn server on port: ", m.port.Num())
if err := firewall.AddInboundRule(m.serviceOptions.Protocol, port.Num()); err != nil {
		return errors.Wrap(err, "failed to add firewall rule")
}
defer func() {
        if err := firewall.RemoveInboundRule(m.serviceOptions.Protocol, port.Num()); err != nil {
		log.Error(logPrefix, "Failed to delete firewall rule for OpenVPN", err)
	}
} ()
```

Reply:
Moving firewall teardown part to the `Serve` as a deferred function looks good to me. Now, regarding the first part, I don't think that:
```go
m.mapPort(m.port.Num())()
```
conveys meaning better than 
```
releasePorts()
```
does. In fact, it looks like an obfuscaton to me. Perhaps there's a better way?

Reply:
Got rid of them. Thanks!